AndroidSDKCore/src/main/java/com/leanplum/internal/Request.java

```diff
@@ -461,18 +464,18 @@ private void sendNow() {
       return;
     }
 
-    this.sendEventually();
+    this.sendEventually(true);
 
     Util.executeAsyncTask(true, new AsyncTask<Void, Void, Void>() {
       @Override
       protected Void doInBackground(Void... params) {
-        sendRequests();
+        sendRequests(true);
```

Comment:
Can you help me understand this logic? It seems like there'd be a race condition, where in the previous `sendEventually` method, we're going to call `sendRequests` in an async task, and then we do it again here in another async task. Why do we need to do this twice, and does it matter if one potentially runs before the other?

Reply:
each of sending async tasks is queued into SingleThreadExecutor, so they will be executed sequentially, not in parallel.